build/helper/metadata_merge_dicts.py

```diff
@@ -6,6 +6,31 @@
 pp = pprint.PrettyPrinter(indent=4, width=80)
 
 
+def merge_helper(metadata, metadata_type, config):
```

Comment:
If you can avoid passing all of config, you should. It looks like only the 'modules' dictionary is needed, which could be optionally None. That way we have less code that depends on the structure of all of config.

Reply:
I prefer to pass all of config. Two reasons.
1. It makes the parameter lists consistent, like I mentioned above
1. When another item from config is needed, it is already available instead of having to go back and change parameter lists all the way back up to where config was last available. Which does happen fairly regularly.

Reply:
Yeah, I can see your point. I just don't like giving a function more information than it needs because it makes it less evident what it is acting on.  It goes against Single Responsibility / Separation of Concerns and it obscures things a little. The fact than in Python you cannot even tell if the thing being passed in is modified or not doesn't help.